Support custom formats in terraform_naming_convention rule (#860)
terraform-linters/tflint: docs/rules/terraform_naming_convention.md

Name | Default | Value
--- | --- | ---
enabled | `false` | Boolean

> format | `snake_case` | `snake_case`, `mixed_snake_case`, `none` or a custom format defined using the `custom_formats` option

Reviewer: ```suggestion
format | `snake_case` | `snake_case`, `mixed_snake_case`, `none` or a custom format defined using the `custom_formats` attribute
```

custom | `""` | String representation of a golang regular expression that the block name must match

> custom_formats | `{}` | Definition of custom formats that can be used in the `format` option

Reviewer: ```suggestion
custom_formats | `{}` | Definition of custom formats that can be used in the `format` attribute
```

The `custom` option defines a custom regex that the identifier must match. This option allows you to have a 
bit more finer-grained control over identifiers, letting you force certain patterns and substrings.

#### `custom_formats`

> The `custom_formats` option defines additional formats that can be used in the `format` option. Like `custom`, it allows

Reviewer: ```suggestion
The `custom_formats` attribute defines additional formats that can be used in the `format` option. Like `custom`, it allows
```

Reviewer: Please also join these lines rather than line wrapping

you to define a custom regular expression that the identifier must match, but it also lets you supply a description
that will be shown when the check fails. Also, it allows you to reuse a custom regex.

> This option consists of a map where the keys are the identifiers of the custom formats, and the values are objects with

Reviewer: ```suggestion
This attribute is a map, where the keys are the identifiers of the custom formats, and the values are objects with
```
